pkg/kubectl/rolling_updater.go

```diff
@@ -67,16 +59,17 @@ type RollingUpdaterConfig struct {
 	// CleanupPolicy defines the cleanup action to take after the deployment is
 	// complete.
 	CleanupPolicy RollingUpdaterCleanupPolicy
-	// UpdateAcceptor is given a chance to accept the new controller after each
-	// scale-up operation. If the controller is accepted, updates continue; if
-	// the controller is rejected, the update will fail immediately.
-	UpdateAcceptor UpdateAcceptor
-	// UpdatePercent is optional; if specified, the amount of replicas scaled up
-	// and down each interval will be computed as a percentage of the desired
-	// replicas for the new RC. If UpdatePercent is nil, one replica will be
-	// scaled up and down each interval. If UpdatePercent is negative, the order
-	// of scaling will be down/up instead of up/down.
+	// UpdatePercent is an optional percentage of replicas to scale up each
+	// interval and is used to compute the minimum pod availability during the
+	// update. If nil, a fixed value of 1 will be used for the scaling increment
+	// and minimum pod availabity. If a negative value is provided, the scaling
+	// direction will be down/up instead of up/down.
 	UpdatePercent *int
+	// UseReadinessAsAcceptance means that the updater will only count ready
+	// pods when determining whether the amount of pods available meets any
+	// minimum pod availability requirement. The default is false, which means
+	// that if a pod exists, it's considered available.
+	UseReadinessAsAcceptance bool
```

Comment:
I think so.  It's a much better check - it'll slow down rolling upgrade
some, but it'll actually make it rolling.

On Thu, Aug 6, 2015 at 4:37 PM, Dan Mace <notifications@github.com> wrote:

> In pkg/kubectl/rolling_updater.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11942#discussion_r36462231>
> :
>
> >  	UpdatePercent *int
> > +	// UseReadinessAsAcceptance means that the updater will only count ready
> > +	// pods when determining whether the amount of pods available meets any
> > +	// minimum pod availability requirement. The default is false, which means
> > +	// that if a pod exists, it's considered available.
> > +	UseReadinessAsAcceptance bool
>
> I'm personally in favor of always using readiness and removing the knob.
> The knob was introduced to maintain the current rolling updater behavior.
> Is the lack of readiness in the master rolling updater code considered a
> bug at this point?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11942/files#r36462231>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift